ratis-server/src/test/java/org/apache/ratis/WatchRequestTests.java

```diff
@@ -457,8 +460,13 @@ static void runTestWatchRequestClientTimeout(TestParameters p) throws Exception
           ex.getCause().getClass());
       if (ex.getCause() != null) {
         if (ex.getCause().getCause() != null) {
-          Assert.assertEquals(TimeoutIOException.class,
-              ex.getCause().getCause().getClass());
+          // when client closed and throw TimeoutException,
+          // RaftClientImpl::handleIOException will random select a leader
+          // because suggested new leader is null.
```

Comment:
I think probbaly, we should ensure there is no retry in the test by setting the attempt count to 1 in the retry policy.. In such case, it will always fail consistently with TimeoutException which is the goal here.

Reply:
@bshashikant Thanks for your review. I have updated.